ENIGMAsystem/SHELL/Platforms/SDL/Makefile

```diff
@@ -7,12 +7,27 @@ override CXXFLAGS += $(shell $(PKG-CONFIG) --cflags sdl2) -DENIGMA_PLATFORM_SDL
 
 ifneq ($(TARGET-PLATFORM), Android)
```

Comment:
instead of putting these if else chains everywhere we could use more consistent file naming and do something like SOURCES+=  /path/$(PLATFORM)/file.cpp

Reply:
The problem here is that SDL is trying to be its own damned operating system, which normally really helps, but ENIGMA will always offer a rough superset of SDL functionality (ie, a superset except for some known missing features available in various other platforms).

That said... this logic reads strangely, yes.

1. This if statement is checking platform != android instead of checking ==. For ease of skimming, it's usually better to use the positive form (ie, remove the not and swap the two bodies).
1. If the system is Android, we include Android/Window.cpp. Otherwise, we include General/....Posix...shell.cpp? What is going on?
1. The general structure of this code is `{ Android }` `{ Windows }` `{ Not Windows { OS X } { Linux } { FreeBSD } }`.
   a. Why is Android not with the other Not-Windows OSes?
   b. Why is only Android including POSIX/shell.cpp? What causes the other POSIX systems to include it?

Reply:
Are you sure $(PLATFORM) will ever equal xlib instead of Linux or FreeBSD?

Reply:
that was just an example and not meant to be taken verbatim. my point was you could avoid the if chain with proper folder names and variables. So if we needed threads.cpp from each platform for sdl you could use SOURCES += $(plat)/threads.cpp so for win32 it would look in win32/threads.cpp and for xlib it would grab xlib/threads.cpp